java/com/facebook/crypto/streams/NativeGCMCipherInputStream.java

```diff
@@ -136,6 +136,6 @@ public synchronized void reset() throws IOException {
 
   @Override
   public long skip(long byteCount) throws IOException {
-    throw new UnsupportedOperationException();
+    return super.skip(byteCount);
```

Comment:
Yes. Thing is super.skip (InputStream.skip) already does that. It creates a byte[2048] an reads on it until end of stream, or byteCount. So it works.

A small improvement could be:
- having our own uniquely instantiated buffer (per stream, so repeated calls don't instantiate more buffers)
- implement read(long) as one of the loops in read *without* the call to System.arraycopy. Maxing the byte count in the buffer size. That way the loop is implemented naturally by the client calling more times to skip when needed (skip, as read, can return less bytes skipped).

But this solution works.